src/vm/analysis/type_checker/tests/mod.rs

```diff
@@ -2529,3 +2530,101 @@ fn test_comparison_types() {
         assert_eq!(expected, &type_check_helper(&bad_test).unwrap_err().err);
     }
 }
+
+#[test]
+fn test_principal_parse() {
+    let good = [
+        // Standard good example.
+        r#"(principal-parse 'STB44HYPYAT2BB2QE513NSP81HTMYWBJP02HPGK6)"#,
```

Comment:
Could we have a test case using a contract id (`'STB44HYPYAT2BB2QE513NSP81HTMYWBJP02HPGK6.contract`) ?

Reply:
Speaking of which, should we have a third component `name: (option (string-ascii))`?

Reply:
Yep, I agree with @lgalabru's suggestion here: a `name` component would be useful for contract principals.

Reply:
OK.. We can add optional third argument.